src/modules/ekf2/EKF/mag_control.cpp

```diff
@@ -302,31 +302,26 @@ bool Ekf::shouldInhibitMag() const
 
 void Ekf::checkMagFieldStrength()
 {
-	if (_params.check_mag_strength) {
-		_control_status.flags.mag_field_disturbed = _NED_origin_initialised
-				? !isMeasuredMatchingGpsMagStrength()
-				: !isMeasuredMatchingAverageMagStrength();
+	if (_mag_data_ready) {
+		if (_params.check_mag_strength
+		    && ((_params.mag_fusion_type <= MAG_FUSE_TYPE_3D) || (_params.mag_fusion_type == MAG_FUSE_TYPE_INDOOR && _control_status.flags.gps))) {
```

Comment:
Why only checking when we're already fusing GPS? Is is expected to fly with GPS in an indoor mode?

Reply:
My understanding of MAG_FUSE_TYPE_INDOOR is that it's a hybrid.

> The same as option 0, but magnetometer or yaw fusion will not be used unless earth frame external aiding (GPS or External Vision) is being used. This prevents inconsistent magnetic fields associated with indoor operation degrading state estimates.